contractcourt/channel_arbitrator.go

```diff
@@ -1135,11 +1146,141 @@ func (c *ChannelArbitrator) sweepAnchors(anchors *lnwallet.AnchorResolutions,
 		if err != nil {
 			return err
 		}
+
+		return nil
+	}
+
+	// Sweep anchors based on different HTLC sets. Notice the HTLC sets may
+	// differ across commitments, thus their deadline values could vary.
+	for htlcSet, htlcs := range c.activeHTLCs {
+		switch {
+		case htlcSet == LocalHtlcSet && anchors.Local != nil:
+			// Inside a local commitment transaction, what's sent
+			// to us (toLocalHTLCs) is the incoming HTLCs, and
+			// what's sent from us (toRemoteHTLCs) is the outgoing
+			// HTLCs.
+			toLocalHTLCs := htlcs.incomingHTLCs
```

Comment:
you might have to check this: IIRC `incoming` here means that it pays to us, regardless of whether this is a local or remote commit.

Reply:
yeah further checked [here](https://github.com/lightningnetwork/lnd/blob/master/lnwallet/channel.go#L2645), got it wrong, will fix.